misc/userscripts/cast

```diff
@@ -133,23 +144,40 @@ echo "jseval -q $(printjs)" >> "$QUTE_FIFO"
 
 tmpdir=$(mktemp -d)
 file_to_cast=${tmpdir}/qutecast
-program_=$(command -v castnow)
+cast_program=$(command -v castnow)
+
+# load optional config
+QUTE_CONFIG_DIR=${QUTE_CONFIG_DIR:-${XDG_CONFIG_HOME:-$HOME/.config}/qutebrowser/}
+QUTE_CAST_CONFIG=${QUTE_CAST_CONFIG:-${QUTE_CONFIG_DIR}/cast_rc}
+if [ -f "$QUTE_CAST_CONFIG" ] ; then
+    source "$QUTE_CAST_CONFIG"
+fi
+
+# if ytdl_program wasn't specified in config, use a fallback
+for p in "$ytdl_program" yt-dlp youtube-dl; do
+  ytdl_program=$(command -v "$p")
+  [ "$ytdl_program" == "" ] || break
+done
 
-if [[ "${program_}" == "" ]]; then
+if [[ "${cast_program}" == "" ]]; then
     msg error "castnow can't be found..."
     exit 1
 fi
+if [[ "${ytdl_program}" == "" ]]; then
+    msg error "youtube-dl (or a drop-in replacement) can't be found..."
```

Comment:
Thanks for the improvements. One nitpick: IMHO it would be nice to include the failing configuration variable value in the error message:

```suggestion
    msg error "youtube-dl (or a drop-in replacement) '${ytdl_program}' can't be found..."
```

Reply:
That would always be empty here, no? Not sure if it makes sense to complicate the logic just for this, though.

Reply:
Sorry, I didn't read the actual condition. In this case I find the erro message misleading: If I see an error

> `"youtube-dl (or a drop-in replacement) can't be found..."`

I'd assume I the configured program was not found on my system, not that no program had been configured in the first place.
So how about:

```suggestion
    msg error "No youtube-dl (or a drop-in replacement) program configured..."
```

Reply:
After looking more closely at the code again ([line159](https://github.com/qutebrowser/qutebrowser/pull/7074/files#diff-34b4768e7037f827d7c3cf3006f417c522b580730cdc1e94506f202c87f7ac83R159)) I think I *finally* get the logic. So I guess the meesage is correct and there is no way to check anymore if ${tydl_program}` was specified at this point. Maybe when switching to an ENVIRONMENT variable, it could be checked and included here, if defined.
Sorry for the noise.